airflow/contrib/hooks/spark_sql_hook.py

```diff
@@ -0,0 +1,150 @@
+        self.name = name
+        self.num_executors = num_executors
+        self.verbose = verbose
+        self.yarn_queue = yarn_queue
+        self.sp = None
+
+    def get_conn(self):
+        pass
+
+    def _prepare_command(self, cmd):
+        """
+        Construct the spark-sql command to execute. Verbose output is enabled
+        as default.
+        :param cmd: command to append to the spark-sql command
+        :type cmd: str
+        :return: full command to be executed
+        """
+        connection_cmd = ["spark-sql"]
+        if self.conf:
+            for conf_el in self.conf.split(","):
+                connection_cmd += ["--conf", conf_el]
+        if self.executor_cores:
+            connection_cmd += ["--executor-cores", self.executor_cores]
+        if self.executor_memory:
+            connection_cmd += ["--executor-memory", self.executor-memory]
+        if self.keytab:
+            connection_cmd += ["--keytab", self.keytab]
+        if self.num_executors:
+            connection_cmd += ["--num_executors", self.num_executors]
+        if self.sql:
+            if self.sql.endswith('.sql'):
+                connection_cmd += ["-f", self.sql]
+            else:
+                connection_cmd += ["-e", self.sql]
+        if self.master:
+            connection_cmd += ["--master", self.master]
+        if self.name:
+            connection_cmd += ["--name", self.name]
+        if self.verbose:
+            connection_cmd += ["--verbose"]
+        if self.yarn_queue:
+            connection_cmd += ["--queue", self.yarn_queue]
+        
+        connection_cmd += cmd
+        logging.debug("Spark-Sql cmd: {} ".format(connection_cmd))
+
+        return connection_cmd
+
+    def run_query(self, cmd="", **kwargs):
+        """
+        Remote Popen (actually execute the Spark-sql query)
+
+        :param cmd: command to remotely execute
+        :param kwargs: extra arguments to Popen (see subprocess.Popen)
+        """
+        prefixed_cmd = self._prepare_command(cmd)
+        self.sp = subprocess.Popen(prefixed_cmd, 
+                                stdout=subprocess.PIPE, 
+                                stderr=subprocess.PIPE)
+        # using two iterators here to support 'real-time' logging 
+        for line in iter(self.sp.stdout.readline, b''):
+            line = line.decode('utf-8').strip()
+            logging.info(line)
+        for line in iter(self.sp.stderr.readline, b''):
+            line = line.decode('utf-8').strip() 
+            logging.info(line)
+        output, stderr = self.sp.communicate()
+
+        if self.sp.returncode:
+            raise AirflowException("Cannot execute {} on {}. Error code is: "
+                                   "{}. Output: {}, Stderr: {}"
+                                   .format(cmd, self.conn.host,
+                                           self.sp.returncode, output, stderr))
+
+    def kill(self):
+        if self.sp:
+            if self.sp.poll() is None:
```

Comment:
I think it would be better to join two expressions:

```python
    def kill(self):
        if self.sp and self.sp.poll() is None:
            ....
```